vars/npmExecuteEndToEndTests.groovy

```diff
@@ -95,12 +100,13 @@ void call(Map parameters = [:]) {
                     withCredentials(credentials) {
                         if (appUrl.parameters) {
                             if (appUrl.parameters instanceof List) {
-                                npmExecuteScripts(script: script, parameters: npmParameters, install: false, virtualFrameBuffer: true, runScripts: [config.runScript], scriptOptions: ["--launchUrl=${appUrl.url}"] + appUrl.parameters)
+                                npmExecuteScripts(script: script, parameters: npmParameters, install: false, virtualFrameBuffer: true, runScripts: [config.runScript], scriptOptions: ["--launchUrl=${appUrl.url}"] + appUrl.parameters, buildDescriptorExcludeList: config.buildDescriptorExcludeList)
                             } else {
                                 error "[${STEP_NAME}] The parameters property is not of type list. Please provide parameters as a list of strings."
                             }
+                        } else {
+                            npmExecuteScripts(script: script, parameters: npmParameters, install: false, virtualFrameBuffer: true, runScripts: [config.runScript], scriptOptions: ["--launchUrl=${appUrl.url}"], buildDescriptorExcludeList: config.buildDescriptorExcludeList)
                         }
-                        npmExecuteScripts(script: script, parameters: npmParameters, install: false, virtualFrameBuffer: true, runScripts: [config.runScript], scriptOptions: ["--launchUrl=${appUrl.url}"])
```

Comment:
Maybe it would be better to have a List "scriptOptions", which defaults to `["--launchUrl=${appUrl.url}"]` and is appended with appUrl.parameters after checking the type, and then npmExecuteScripts is called only once with that List.

Reply:
Good idea I'll address it!